src/VisualStudio/Core/Def/Implementation/VisualStudioMetadataAsSourceFileSupportService.cs

```diff
@@ -1,76 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-#nullable disable
-
-using System;
-using System.ComponentModel.Composition;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
-using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.MetadataAsSource;
-using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.Shell.Interop;
-
-namespace Microsoft.VisualStudio.LanguageServices.Implementation;
-
-[Export(typeof(VisualStudioMetadataAsSourceFileSupportService))]
-internal sealed class VisualStudioMetadataAsSourceFileSupportService : IVsSolutionEvents
-{
-    private readonly IThreadingContext _threadingContext;
-    private readonly IMetadataAsSourceFileService _metadataAsSourceFileService;
-
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public VisualStudioMetadataAsSourceFileSupportService(
-        IThreadingContext threadingContext,
-        IMetadataAsSourceFileService metadataAsSourceFileService)
-    {
-        _threadingContext = threadingContext;
-        _metadataAsSourceFileService = metadataAsSourceFileService;
-    }
-
-    public async Task InitializeAsync(IAsyncServiceProvider serviceProvider, CancellationToken cancellationToken)
-    {
-        var solution = await serviceProvider.GetServiceAsync<SVsSolution, IVsSolution>(_threadingContext.JoinableTaskFactory).ConfigureAwait(false);
-        await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-        // Intentionally ignore the event-cookie we get back out.  We never stop listening to solution events.
-        ErrorHandler.ThrowOnFailure(solution.AdviseSolutionEvents(this, out _));
-    }
-
-    public int OnAfterCloseSolution(object pUnkReserved)
-    {
-        _metadataAsSourceFileService.CleanupGeneratedFiles();
```

Comment:
this existed to hear about solution close, so we could clean things up.  but we can already just clean things up when we startup anyways.